pkg/certificate/mrc_reconciler.go

```diff
@@ -130,7 +131,12 @@ func shouldUpdateState(mrc *v1alpha2.MeshRootCertificate) bool {
 // update the MRCComponentStatus
 func (m *Manager) updateMRCComponentStatus(mrc *v1alpha2.MeshRootCertificate) error {
 	// TODO(5046): determine the status to set the owned components to.
-	var status v1alpha2.MeshRootCertificateComponentStatus
+	// default component status is Unknown
+	var status v1alpha2.MeshRootCertificateComponentStatus = constants.MRCComponentStatusUnknown
+
+	if shouldChangeStatusToValidating(mrc) {
```

Comment:
I wonder if we can create a generic implementation for,

`getNextStatus`, which simply returns the next status we should set it too, instead of having 1 per status. Might not be able to, but just a thought